lxd/api_project.go

```diff
@@ -952,6 +953,48 @@ func projectStateGet(d *Daemon, r *http.Request) response.Response {
 	return response.SyncResponse(true, &state)
 }
 
+// uplinkIPLimitValidator returns a validator function for uplink IP limits.
+// The protocol argument specifies whether we should validate ipv4 or ipv6.
+func uplinkIPLimitValidator(s *state.State, projectName string, networkName string, protocol string) func(string) error {
+	return func(value string) error {
+		// Perform cheaper checks on the value first.
+		providedIPQuota, err := strconv.Atoi(value)
+		if err != nil {
+			return err
+		}
+
+		if providedIPQuota < 0 {
+			return fmt.Errorf("Value must be non-negative")
+		}
+
+		// Populate quota variables accordingly.
+		IPV4AddressQuota := -1
+		IPV6AddressQuota := -1
+
+		if protocol == "ipv6" {
+			IPV6AddressQuota = providedIPQuota
+		}
+
+		if protocol == "ipv4" {
+			IPV4AddressQuota = providedIPQuota
+		}
+
+		// Check if the provided value is equal or lower to the number of uplink addresses currently in use
+		// on the provided project and in the specified network.
+		// We are only interested on the result for the desired protocol, the other will always come out as true.
+		invalidIPV4Quota, invalidIPV6Quota, err := limits.UplinkAddressQuotasExceeded(s, projectName, networkName, IPV4AddressQuota, IPV6AddressQuota)
+		if err != nil {
+			return err
+		}
+
+		if protocol == "ipv4" && invalidIPV4Quota || protocol == "ipv6" && invalidIPV6Quota {
+			return fmt.Errorf("Value %s is below current number of used uplink addresses", value)
```

Comment:
We use %q for user inputted strings

Reply:
Also "Value" here is a bit vague, can we say "Uplink IPv4 limit %q is below current number of used uplink address"